File: 프로그래머스/lv1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.py
def solution(left, right):
    answer = 0
    
    def cal(number):
        cnt = 0
        for i in range(1, number + 1):
            if number % i == 0:
                cnt += 1
        return cnt
    
    for i in range(left, right + 1):
        if cal(i) % 2 == 0:
            answer += i
        else:
            answer -= i
    
    return answer

File: 프로그래머스/1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.js
function solution(left, right) {
    var answer = 0;
    const divisorCounter = (num) => {
        let count = 0
        for (let i = 1; i <= Math.sqrt(num); i++) {
            if (num % i === 0) {
                if (i === Math.sqrt(num)) count++
                else count += 2
            }
        }
        return count
    }
    
    for (let i = left; i <= right; i++) {
        if (divisorCounter(i) % 2 === 0) answer += i
        else answer -= i
    }
    return answer;
}